README
NAME

    OpenGbg - An interface to the Open Data API of Gothenburg

VERSION

    Version 0.1301, released 2015-01-16.

SYNOPSIS

        use OpenGbg;
    
        my $opengbg = OpenGbg->new(key => 'secret-api-key');
    
        $response = $opengbg->styr_och_stall->get_bike_stations;
    
        print $response->stations->get_by_index(0)->to_text;

DESCRIPTION

    OpenGbg is a way to connect to and use the open data published by the
    city of Gothenburg <https://en.wikipedia.org/wiki/Gothenburg>.

    The open data homepage is located at http://data.goteborg.se/. All
    official documentation is in Swedish, but the license agreement is
    published in English here <https://gbgdata.wordpress.com/goopen/>.

    To use the API you need to get a free api key.

METHOD

 new()

    Takes an optional key-value pair, the key is key and the value your api
    key, see authenticate.

    Returns a OpenGbg::Handler object on which you then call the services.

AUTHENTICATE

    Once you have your api key you can use it in two different ways:

    1. You can give it in the constructor:

        my $opengbg = OpenGbg->new(key => 'secret-api-key');

    2. You can save it in a file named .opengbg.ini in your homedir:

        [API]
        key = secret-api-key

SERVICES

    The following services are currently implemented in this distribution:

    AirQuality - Data on air quality

    Bridge - Data on the openness of Göta Älvbron

    StyrOchStall - Data on rent-a-bike stations

    TrafficCamera - Data on traffic cameras, and their images

NAMING

    Most names related to the services are de-camelized, while others are
    lower-cased (no underscores). For example, the service
    'GetBikeStations' is called like this:

        my $gbg = OpenGbg->new;
        my $stations = $gbg->get_bike_stations;

    All calls to services are prefixed with 'get' even if the service isn't
    named that way. On the other hand, the 'service' suffix on some
    services are removed.

    Date/time intervals are always called 'start' and 'end' (in the web
    services they are sometimes called 'start' and 'stop').

DISCLAIMER

    This is not an official distribution.

SOURCE

    https://github.com/Csson/p5-OpenGbg

HOMEPAGE

    https://metacpan.org/release/OpenGbg

AUTHOR

    Erik Carlsson <info@code301.com>

COPYRIGHT AND LICENSE

    This software is copyright (c) 2015 by Erik Carlsson.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.



Changes
Revision history for OpenGbg

{{$NEXT}}
 [API Changes]

 [Bug Fixes]

 [Documentation]

 [Enhancements]

 [New Features]

 [Other Changes]

 [Requirements]

 [Stats & Dependency Changes]

 [Testing]

0.1404 2018-05-19T14:44:40Z UTC
 [API Changes]
 - Updated ::TrafficCamera to API version 0.2, which adds lat/long
   information to the camera devices.

 [Stats & Dependency Changes]
 - 23 files changed, 41 insertions(+), 25 deletions(-)
 - (dev req) ~ Dist::Zilla 6.010 --> 6.012
 - (dev req) ~ Dist::Zilla::Plugin::Git 2.041 --> 2.043
 - (dev sug) ~ Dist::Iller::Config::Author::CSSON 0.0320 --> 0.0321

0.1403 2018-03-14T18:26:16Z UTC
 [Documentation]
 - Updated link to the official license agreement.

 [New Features]
 - Added a couple of shortcut methods in the traffic camera service.

 [Other Changes]
 - The OpenGbg::StyrOchStall namespace is deprecated, due to the
   removal/renaming of the underlying web service.
 - Types::DateTime 0.002 fixed the previously mentioned issue, so the
   custom DateTime type class has been removed.

 [Stats & Dependency Changes]
 - 26 files changed, 61 insertions(+), 46 deletions(-)
 - (run req) + Types::DateTime 0.002
 - (run req) ~ Type::Tiny 1.000005 --> 1.000006
 - (dev req) ~ Dist::Zilla 5.000 --> 6.010
 - (dev req) ~ Dist::Zilla::Plugin::BumpVersionAfterRelease::Transitional
   0.007 --> 0.008
 - (dev req) ~ Dist::Zilla::Plugin::ChangeStats::Dependencies::Git 0.0100
   --> 0.0200
 - (dev req) ~ Dist::Zilla::Plugin::CopyFilesFromBuild 0.151680 -->
   0.170880
 - (dev req) ~ Dist::Zilla::Plugin::Git 2.036 --> 2.041
 - (dev req) ~ Dist::Zilla::Plugin::Git::CheckFor::CorrectBranch 0.013 -->
   0.014
 - (dev req) ~ Dist::Zilla::Plugin::Git::Contributors 0.017 --> 0.030
 - (dev req) ~ Dist::Zilla::Plugin::MetaProvides::Package 2.003001 -->
   2.004003
 - (dev req) ~ Dist::Zilla::Plugin::PromptIfStale (any) --> 0.054
 - (dev req) ~ Dist::Zilla::Plugin::ReadmeAnyFromPod 0.150250 --> 0.163250
 - (dev req) ~ Dist::Zilla::Plugin::Test::Compile 2.054 --> 2.057
 - (dev req) ~ Dist::Zilla::Plugin::Test::ReportPrereqs 0.021 --> 0.027
 - (dev sug) ~ Dist::Iller::Config::Author::CSSON 0.0312 --> 0.0320

0.1402 2016-08-12T22:07:56Z UTC
 [Bug Fixes]
 - Types::DateTime is (currently) broken with recent releases of
   DateTime::Locale, so add a custom DateTime type for now.

 [Stats & Dependency Changes]
 - 25 files changed, 43 insertions(+), 31 deletions(-)
 - (run req) ~ DateTime 1.05 --> 1.36
 - (run req) ~ Moose 2.1400 --> 2.1803
 - (run req) - Types::DateTime
 - (dev req) + Dist::Zilla::Plugin::PromptIfStale (any)
 - (dev sug) ~ Dist::Iller::Config::Author::CSSON 0.0309 --> 0.0312

0.1401 2016-05-05T10:21:46Z UTC
 [API Changes]
 - Deprecated 'measurement' in
   OpenGbg::Service::AirQuality::GetLatestMeasurement. All methods and
   attributes in OpenGbg::Service::AirQuality::Measurement are now
   available directly on ::GetLatestMeasurement.

 [Enhancements]
 - Added convenience method 'is_open' to OpenGbg::Service::Bridge.

 [Stats & Dependency Changes]
 - 30 files changed, 167 insertions(+), 61 deletions(-)
 - (run req) ~ DateTime (any) --> 1.05
 - (run req) ~ MooseX::AttributeShortcuts (any) --> 0.028
 - (run req) ~ Path::Tiny (any) --> 0.072
 - (run req) ~ Type::Tiny (any) --> 1.000005
 - (run req) ~ Types::Path::Tiny (any) --> 0.005
 - (run req) ~ namespace::autoclean 0.18 --> 0.22
 - (run req) ~ perl 5.010001 --> 5.010000
 - (dev req) + Badge::Depot 0.0103
 - (dev req) + Badge::Depot::Plugin::Coverage (any)
 - (dev req) + Badge::Depot::Plugin::Cpantesters (any)
 - (dev req) + Badge::Depot::Plugin::Kwalitee (any)
 - (dev req) + Badge::Depot::Plugin::Perl (any)
 - (dev req) + Badge::Depot::Plugin::Travis (any)
 - (dev req) + Dist::Zilla 5.000
 - (dev req) + Dist::Zilla::Plugin::Authority 1.009
 - (dev req) + Dist::Zilla::Plugin::BumpVersionAfterRelease::Transitional
   0.007
 - (dev req) + Dist::Zilla::Plugin::ChangeStats::Dependencies::Git 0.0100
 - (dev req) + Dist::Zilla::Plugin::Git 2.036
 - (dev req) + Dist::Zilla::Plugin::Git::CheckFor::CorrectBranch 0.013
 - (dev req) + Dist::Zilla::Plugin::Git::Commit (any)
 - (dev req) + Dist::Zilla::Plugin::MetaConfig (any)
 - (dev req) + Dist::Zilla::Plugin::NextRelease::Grouped 0.0200
 - (dev req) + Dist::Zilla::Plugin::RewriteVersion::Transitional 0.007
 - (dev req) + Pod::Weaver::Section::GenerateSection 1.01
 - (dev req) + Test::EOF (any)
 - (dev req) + Test::Version 1
 - (dev req) ~ Dist::Zilla::Plugin::CheckChangesHasContent (any) --> 0.008
 - (dev req) ~ Dist::Zilla::Plugin::Clean (any) --> 0.07
 - (dev req) ~ Dist::Zilla::Plugin::CopyFilesFromBuild (any) --> 0.151680
 - (dev req) ~ Dist::Zilla::Plugin::Git::Contributors (any) --> 0.017
 - (dev req) ~ Dist::Zilla::Plugin::GithubMeta (any) --> 0.54
 - (dev req) ~ Dist::Zilla::Plugin::InstallRelease (any) --> 0.008
 - (dev req) ~ Dist::Zilla::Plugin::MetaProvides::Package (any) -->
   2.003001
 - (dev req) ~ Dist::Zilla::Plugin::PodWeaver (any) --> 4.006
 - (dev req) ~ Dist::Zilla::Plugin::ReadmeAnyFromPod (any) --> 0.150250
 - (dev req) ~ Dist::Zilla::Plugin::RunExtraTests (any) --> 0.028
 - (dev req) ~ Dist::Zilla::Plugin::Test::Compile (any) --> 2.054
 - (dev req) ~ Dist::Zilla::Plugin::Test::EOF (any) --> 0.0501
 - (dev req) ~ Dist::Zilla::Plugin::Test::EOL (any) --> 0.18
 - (dev req) ~ Dist::Zilla::Plugin::Test::Kwalitee::Extra (any) --> v0.2.1
 - (dev req) ~ Dist::Zilla::Plugin::Test::NoTabs (any) --> 0.15
 - (dev req) ~ Dist::Zilla::Plugin::Test::ReportPrereqs (any) --> 0.021
 - (dev req) ~ Dist::Zilla::Plugin::Test::Version (any) --> 1.05
 - (dev req) ~ Dist::Zilla::Plugin::TravisYML (any) --> 1.15
 - (dev req) ~ Pod::Weaver::Section::Badges (any) --> 0.0401
 - (dev req) ~ Test::More (any) --> 0.88
 - (dev req) ~ Test::NoTabs (any) --> 1.4
 - (dev req) ~ Test::Warnings (any) --> 0.026
 - (dev req) - Dist::Iller::Config::Author::CSSON
 - (dev req) - Dist::Zilla::Plugin::MetaProvides::Class
 - (dev req) - Dist::Zilla::Plugin::NextRelease

Makefile.PL
# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v6.012.
use strict;
use warnings;

use 5.010000;

use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "An interface to the Open Data API of Gothenburg",
  "AUTHOR" => "Erik Carlsson <info\@code301.com>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "OpenGbg",
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.010000",
  "NAME" => "OpenGbg",
  "PREREQ_PM" => {
    "Config::Any" => 0,
    "DateTime" => "1.36",
    "DateTime::Format::HTTP" => 0,
    "File::HomeDir" => 0,
    "HTTP::Tiny" => 0,
    "Moose" => "2.1803",
    "MooseX::AttributeShortcuts" => "0.028",
    "Path::Tiny" => "0.072",
    "Sub::Exporter" => 0,
    "Syntax::Feature::Qs" => "0.2004",
    "Throwable::SugarFactory" => "0.152700",
    "Type::Tiny" => "1.000006",
    "Types::DateTime" => "0.002",
    "Types::Path::Tiny" => "0.005",
    "XML::Rabbit" => "0.4.1",
    "namespace::autoclean" => "0.22",
    "syntax" => 0
  },
  "TEST_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Test::More" => "0.96"
  },
  "VERSION" => "0.1405",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "Config::Any" => 0,
  "DateTime" => "1.36",
  "DateTime::Format::HTTP" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::HomeDir" => 0,
  "File::Spec" => 0,
  "HTTP::Tiny" => 0,
  "IO::Handle" => 0,
  "IPC::Open3" => 0,
  "Moose" => "2.1803",
  "MooseX::AttributeShortcuts" => "0.028",
  "Path::Tiny" => "0.072",
  "Sub::Exporter" => 0,
  "Syntax::Feature::Qs" => "0.2004",
  "Test::More" => "0.96",
  "Throwable::SugarFactory" => "0.152700",
  "Type::Tiny" => "1.000006",
  "Types::DateTime" => "0.002",
  "Types::Path::Tiny" => "0.005",
  "XML::Rabbit" => "0.4.1",
  "namespace::autoclean" => "0.22",
  "syntax" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);
